runtime/shutdown/shutdown.go

```diff
@@ -0,0 +1,87 @@
+package shutdown
+
+import "context"
+
+// Progress provides progress information about an ongoing graceful shutdown process.
+//
+// The process broadly consists of  phases:
+//
+// 1. Drain active handlers
+//
+// As soon as the graceful shutdown process is initiated, the service will stop accepting new
+// incoming API calls and Pub/Sub messages. It will continue to process already running handlers
+// until they complete (or the ForceCloseTasks deadline is reached).
+//
+// This phase continues until all active handlers have completed or the ForceCloseTasks deadline
+// is reached, whichever happens first. The OutstandingRequests, OutstandingPubSubMessages, and
+// OutstandingTasks contexts provide insight into what tasks are still active.
+//
+// 2. Shut down infrastructure resources
+//
+// When all active handlers have completed, Encore will begin shutting down infrastructure resources.
+// Encore closes all open database connections, cache connections.
```

Comment:
Based on our conversation from yesterday, shouldn't this read:

```suggestion
// When all active handlers have completed and all your shutdown [Handler.Shutdown] functions have returned, Encore will begin shutting down infrastructure resources.
// Encore closes all open database connections, cache connections.
```

Reply:
It's saying the same thing ("active handlers" refers to running [Handler.Shutdown] functions), but I can make it a bit more precise.

Reply:
Given this line above, it's not clear:

```
// As soon as the graceful shutdown process is initiated, the service will stop accepting new
// incoming API calls and Pub/Sub messages. It will continue to process already running handlers
// until they complete
```